### PR TITLE
[llvm] Specialize element shape for LLVM backend

### PR DESCRIPTION
**Pull request by turbo0628**
Related issue = #5044

### CODE REVIEW

File: taichi/codegen/codegen_llvm.cpp [hunk 1]
```diff
@@ -1623,8 +1640,15 @@ void CodeGenLLVM::visit(ExternalPtrStmt *stmt) {
       llvm::PointerType::get(tlctx->get_data_type(dt), 0));
 
   auto linear_index = tlctx->get_constant(0);
+  int element_shape_idx = 0;
   for (int i = 0; i < num_indices; i++) {
```
Comment (k-ye) [inline taichi/codegen/codegen_llvm.cpp:1644]: Just to confirm. In the future `num_indices` should always be equal to `len(array_shape)`, right? @ailzhang @turbo0628

**Review by k-ye — APPROVED** [review]
LGTM!

File: taichi/codegen/codegen_llvm.cpp (hunk 1 above)
Comment (turbo0628) [reply taichi/codegen/codegen_llvm.cpp:1644]: If array_shape excludes element_shape, then yes. I plan to remove element shape from the `indices` vector in a follow-up PR.

Comment (ailzhang) [reply taichi/codegen/codegen_llvm.cpp:1644]: Yup that's exactly the goal of these refactors.

Comment (k-ye) [reply taichi/codegen/codegen_llvm.cpp:1644]: Cool!

**Pull request merged**